### DISCUSSION TITLE
No module named 'floris.tools'

### DISCUSSION THREAD

**Discussion by zkdtxby** [NatLabRockies/floris#908] [Q&A]
I get a "No module named 'floris.tools'" error when using "from floris.tools import FlorisInterface" in jupyter notebook.Can you help me analyze the reason? Thank you very much!
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[3], line 11
      8 from typing import Union
     10 import floris
---> 11 from floris.tools import FlorisInterface
     12 from floris.tools.visualization import visualize_cut_plane
     13 from floris.tools.visualization import plot_turbines_with_fi

ModuleNotFoundError: No module named 'floris.tools'
![image](https://github.com/NREL/floris/assets/84053659/080228a7-5f3a-4044-b2eb-f0399e52393d)

**Comment by QW543** [comment]
I encountered the same problem as well, may I ask if you have resolved it?

**Comment by misi9170** [comment]
Hi @QW543 ,

Which version of FLORIS are you using? `floris.tools` was a module in FLORIS v3 (last release was [v3.6](https://github.com/NREL/floris/releases/tag/v3.6) in April 2024). Since the release of FLORIS v4 (latest release [v4.4.2](https://github.com/NREL/floris/releases/tag/v4.4.2) earlier this month), there is no longer a `floris.tools` module and `FlorisInterface` has been replaced by `FlorisModel`. 

See [here](https://nrel.github.io/floris/v3_to_v4.html) for more details about v4 compared to v3.

Misha
